WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/EmptyState.swift

```diff
@@ -9,30 +9,38 @@ struct EmptyState: View {
     @State var buttonAction: (() -> Void)?
 
     var body: some View {
-        VStack(spacing: Constants.verticalSpacing) {
+        VStack {
+            if let image = image {
+                // Downscale the bigger images to the max width
+                ViewThatFits {
+                    Image(uiImage: image)
+
+                    Image(uiImage: image)
+                        .resizable()
+                        .scaledToFit()
+                }
```

Comment:
There was a discrepancy between the SwiftUI `EmptyState` and the UIKit one; here, we were forcing a fixed width on the illustrations. This is problematic with the new illustrations as they are intended to be smaller, so I had to either remove the frame constraint completely (which would have an impact on existing screens), or use a different approach, so I experimented with having the width configurable and with having a `maxWidth` to keep the UI the same when having larger illustrations.

I went with the second approach, but I had to use this dance here to get SwiftUI to avoid resizing the image unless it's bigger than the maxWidth, this is based on this [SO reply](https://stackoverflow.com/a/78285211), if there is an easier way that I'm missing, please let me know.

Reply:
Nice, this is simpler and works well, thanks @itsmeichigo 🙏